analyzer/src/namespace/scopes.rs

```diff
@@ -31,10 +38,12 @@ pub struct ContractFieldDef {
 #[derive(Clone, Debug, PartialEq)]
 pub struct ModuleScope {
     pub type_defs: HashMap<String, Type>,
+    contract_defs: HashMap<String, ContractDef>,
```

Comment:
As mentioned below, I think we could get rid of this field by adding the contract's type right away in `contract_def`